DB/src/main/java/io/deephaven/db/v2/parquet/ParquetInstructions.java

```diff
@@ -1,63 +1,91 @@
 package io.deephaven.db.v2.parquet;
 
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
-public class ParquetInstructions {
-    private Map<String, String> columnNameMapping;
-    private Map<String, String> reverseColumnNameMapping;
-
+public abstract class ParquetInstructions {
     public ParquetInstructions() {
     }
 
-    public String getColumnNameMapping(final String fromName) {
-        if (columnNameMapping == null) {
-            return null;
-        }
-        return columnNameMapping.get(fromName);
+    public final String getColumnNameFromParquetColumnNameOrDefault(final String parquetColumnName) {
+        final String mapped = getColumnNameFromParquetColumnName(parquetColumnName);
+        return (mapped != null) ? mapped : parquetColumnName;
     }
+    public abstract String getParquetColumnNameFromColumnNameOrDefault(final String columnName);
+    public abstract String getColumnNameFromParquetColumnName(final String parquetColumnName);
 
-    public Map<String, String> getColumnNameMappings() {
-        if (columnNameMapping == null) {
-            return Collections.emptyMap();
+    public static final ParquetInstructions EMPTY = new ParquetInstructions() {
+        @Override
+        public String getParquetColumnNameFromColumnNameOrDefault(final String columnName) {
+            return columnName;
         }
-        return columnNameMapping;
-    }
+        @Override
+        public String getColumnNameFromParquetColumnName(final String parquetColumnName) {
+            return null;
+        }
+    };
 
-    public Map<String, String> getReverseColumnNameMappings() {
-        if (reverseColumnNameMapping == null) {
-            return Collections.emptyMap();
+    private static class Base extends ParquetInstructions {
+        private Map<String, ColumnInstructions> columnNameToInstructions;
```

Comment:
I don't really understand the lack of symmetry between these two maps.

Reply:
I reworked to use KeyedObjectHashMap and make "more" symmetric now.  Note it is still not fully symmetric because I don't want to keep entries in the parquet columns map when the mapping is default (same as DH column name).